src/System.Memory/src/System/Buffers/Text/Utf8Formatter/FormattingHelpers.cs

```diff
@@ -189,6 +215,44 @@ public static int CountDigits(ulong value)
             return digits;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int CountDigits(uint value)
+        {
+            int digits = 1;
+            if (value >= 1000)
+            {
+                if (value >= 1000000)
+                {
+                    value = value / 1000000;
+                    digits += 6;
+                }
+                else
+                {
+                    value = value / 1000;
+                    digits += 3;
+                }
+            }
+            if (value < 10) 
+            { 
+                // no-op
+            }
+            else if (value < 100) 
+            {
+                digits += 1;
+            }
+            else if (value < 1000)
+            {
+                digits += 2;
+            }
+            else
+            {
+                Debug.Assert(value < 1000);
```

Comment:
This Debug.Assert is incorect. It should be `Debug.Assert(value < 10000);`

**Question:** How come the CI is green (specifically for debug builds). We have tests that cover this scenario and when I run the tests locally, I fail fast due to Debug.Assert being false.
```text
     at System.Environment.FailFast(System.String, System.Exception)
     at System.Diagnostics.Debug.ShowAssertDialog(System.String, System.String, System.String)
     at System.Diagnostics.Debug.Assert(Boolean, System.String, System.String)
     at System.Diagnostics.Debug.Assert(Boolean)
     at System.Buffers.Text.FormattingHelpers.CountDigits(UInt32)
     at System.Buffers.Text.Utf8Formatter.TryFormatUInt64LessThanBillionMaxUInt(UInt64, System.Span`1<Byte>, Int32 ByRef)
     at System.Buffers.Text.Utf8Formatter.TryFormatUInt64Default(UInt64, System.Span`1<Byte>, Int32 ByRef)
     at System.Buffers.Text.Utf8Formatter.TryFormatUInt64(UInt64, System.Span`1<Byte>, Int32 ByRef, System.Buffers.StandardFormat)
     at System.Buffers.Text.Utf8Formatter.TryFormat(UInt64, System.Span`1<Byte>, Int32 ByRef, System.Buffers.StandardFormat)
```

Reply:
Same with the Debug.Assert in TryFormatInt64MoreThanNegativeBillionMaxUInt since Quintilion contains 19 digits, not 18.
```C#
uint OverNineDigits = (uint)(Quintilion / Utf8Constants.Billion);
int digitCountOverNineDigits = FormattingHelpers.CountDigits(OverNineDigits);
// Why didn't this fail?
Debug.Assert(digitCountOverNineDigits >= 1 && digitCountOverNineDigits <= 9);

// The Debug.Assert should be as follows:
Debug.Assert(digitCountOverNineDigits >= 1 && digitCountOverNineDigits <= 10);
```

Reply:
> Question: How come the CI is green (specifically for debug builds). We have tests that cover this scenario and when I run the tests locally, I fail fast due to Debug.Assert being false.

Which test(s)?

Reply:
We test all powers of 10.
See https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/ParsersAndFormatters/TestData.cs#L145